onnxoptimizer/passes/eliminate_nop_flatten.h

```diff
@@ -0,0 +1,56 @@
+// ATTENTION: The code in this file is highly EXPERIMENTAL.
+// Adventurous users should note that the APIs will probably change.
+
+#pragma once
+
+#include "onnxoptimizer/pass.h"
+
+namespace ONNX_NAMESPACE {
+namespace optimization {
+
+struct EliminateNopFlatten final : public PredicateBasedPass {
+  explicit EliminateNopFlatten()
+      : PredicateBasedPass(PassType::Nop, PassEfficiency::Complete,
+                           PassOptimizationType::Compute) {}
+
+  std::string getPassName() const override { return "eliminate_nop_flatten"; }
+
+  bool patternMatchPredicate(Node *node) override {
+    if (node->kind() != Symbol("Flatten")) {
+      return false;
+    }
+    const Value *input = node->input();
+    if (!input->has_sizes()) {
+      return false;
+    }
+    const auto input_shape = input->sizes();
+    const int axis = node->hasAttribute(kaxis) ? node->i(kaxis) : 1;
+    if (input_shape.size() == 2) {
```

Comment:
I think we could loose the limit from `input_shape.size() == 2` to N due to def of `Flatten` support `axis = [-r, r]`.